### PR TITLE
Clean up some code

### PR DESCRIPTION
**Pull request by marcpopMSFT**
Check the unzip path before extracting to it
Switch to XxHash3 and XxHash64 which is for non-security usage.

### CODE REVIEW

File: src/Installer/core-sdk-tasks/ExtractArchiveToDirectory.cs [hunk 1]
```diff
@@ -121,8 +130,17 @@ public override bool Execute()
                                     entryName = entryName.StartsWith("./") ? entryName[2..] : entryName;
                                     if (ShouldExtractItem(entryName))
                                     {
-                                        Log.LogMessage(entryName);
                                         string destinationPath = Path.Combine(DestinationDirectory, entryName);
+                                        string destinationFileName = Path.GetFullPath(destinationPath);
+                                        string fullDestDirPath = Path.GetFullPath(DestinationDirectory + Path.DirectorySeparatorChar);
+
+                                        if (!destinationFileName.StartsWith(fullDestDirPath)) {
```
Comment (GrabYourPitchforks) [inline src/Installer/core-sdk-tasks/ExtractArchiveToDirectory.cs:137]: Unless you know that both paths end with a slash, this is not a secure way of performing a path prefix check.

**Review by javiercn — APPROVED** [review]
Looks good for the Blazor changes.

**Comment by marcpopMSFT** [comment]
@dotnet/product-construction I'm getting VMR synchronization failures in this PR in the VMR and source build legs. Any ideas?
`Failed to apply the patch root-95dad63-298b71f-1.patch to /.`

**Comment by MichaelSimons** [comment]
> @dotnet/product-construction I'm getting VMR synchronization failures in this PR in the VMR and source build legs. Any ideas? `Failed to apply the patch root-95dad63-298b71f-1.patch to /.`

@premun

**Comment by premun** [comment]
@MichaelSimons I suspect it might be the readme after all. I forgot we're no longer generating on the fly from the template as we moved the component list into its own file. I reverted the README changes in the VMR

File: src/Installer/core-sdk-tasks/ExtractArchiveToDirectory.cs [hunk 2]
```diff
@@ -170,6 +181,26 @@ public override bool Execute()
             return retVal;
         }
 
+        private string GetFullDirectoryPathWithSeperator(string directory)
+        {
+            string fullDirectoryPath = Path.GetFullPath(directory);
+
+            if (!fullDirectoryPath.EndsWith(Path.DirectorySeparatorChar.ToString()))
```
Comment (jaredpar) [inline src/Installer/core-sdk-tasks/ExtractArchiveToDirectory.cs:188]: `EndsWith(string)` should specify a comparison. A bit surprised at the `ToString()` allocation. Why not just look at the last character? 

```csharp
if (fullDirectoryPath.Length > 0 && fullDirectoryPath[^1] == Path.DirectorySeparatorChar)
```

**Comment by marcpopMSFT** [comment]
Appears to be hitting the new known templating test issue.

File: src/Installer/core-sdk-tasks/ExtractArchiveToDirectory.cs (hunk 2 above)
Comment (marcpopMSFT) [reply src/Installer/core-sdk-tasks/ExtractArchiveToDirectory.cs:188]: I was told to copy the code from here: https://github.com/dotnet/runtime/blob/main/src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchiveEntry.Extract.cs#L111 

It failed without the ToString(). Your option is probably better so I'll try that.

Comment (marcpopMSFT) [reply src/Installer/core-sdk-tasks/ExtractArchiveToDirectory.cs:188]: @jaredpar this builds for netfx and your code doesn't compile for netfx.

Comment (jaredpar) [reply src/Installer/core-sdk-tasks/ExtractArchiveToDirectory.cs:188]: The joys of NetFX ... :( 

Should at least do the explicit string comparison part.

**Review by joeloff — APPROVED** [review]

**Comment by marcpopMSFT** [comment]
Suppression was approved.

**Pull request merged**